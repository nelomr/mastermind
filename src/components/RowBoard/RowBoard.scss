.row-board {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  opacity: .5;
  pointer-events: none;
  transition: opacity .3s ease;

  @include breakpoint($min-desktop) {
    margin-bottom: 40px;
  }

  &.is-current {
    opacity: 1;
    pointer-events: auto;
  }

  &:last-child {
    margin-bottom: 0;
  }
}

.row-board--index {
  font-size: 18px;
  font-weight: bold;
}

.row-board--guesses {
  display: flex;
  justify-content: space-around;
  flex: 1;

  @include breakpoint($min-desktop) {
    padding: 0 30px 0 60px;
  }
}