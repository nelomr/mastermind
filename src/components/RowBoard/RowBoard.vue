<template>
  <div class="row-board" :class="{'is-current': indexRow === row}">
    <div class="row-board--index" v-text="setIndexRow" />
    <div class="row-board--guesses">
      <ColorButton 
        v-for="(color, index) in numSlots"
        :key="'color'+index"
        :isGuessButton="true"
        :index="index"
        :indexRow="indexRow"
      />
    </div>
    <PegsList 
      :numSlots="numSlots"
      :indexRow="indexRow"
    />
  </div>
</template>

<script>
import { mapState } from 'vuex';
import ColorButton from '@/components/ColorButton/ColorButton';
import PegsList from '@/components/PegsList/PegsList';

export default {
  name: 'RowBoard',
  props: {
    indexRow: {
      type: Number,
      default: 0
    }
  },
  components: {
    ColorButton,
    PegsList
  },
  computed: {
    ...mapState({
      numSlots: state => state.defaultConfig.num_slots,
      row: 'row'
    }),
    setIndexRow() {
      return this.indexRow < 9 ? `0${this.indexRow+1}`: this.indexRow+1
    }
  }
}
</script>
<style lang="scss">
  @import "./RowBoard.scss";
</style>