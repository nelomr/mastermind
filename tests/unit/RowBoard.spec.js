import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import RowBoard from '@/components/RowBoard/RowBoard.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('RowBoard.vue', () => {
  let defaultConfig = {'num_slots': 4};
  let row = 0;
  let indexRow = 0;

  let props;
  let store;
  let state;

  beforeEach(() => {
    props = {
      indexRow
    };
    state = {
      defaultConfig,
      row,
    };
    store = new Vuex.Store({
      state
    });
  });
  const build = () => {
    const wrapper = shallowMount(RowBoard, {
      props,
      mocks: {$store: store}
    });

    return {
        wrapper
    };
  };

  it('renders the component list', () => {
    const {wrapper} = build();

    expect(wrapper.html()).toMatchSnapshot();
  });
})
