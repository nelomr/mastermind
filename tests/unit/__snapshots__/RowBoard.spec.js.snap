// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowBoard.vue renders the component list 1`] = `
<div class="row-board">
  <div class="row-board--index">NaN</div>
  <div class="row-board--guesses">
    <colorbutton-stub color="lightgray" index="0" indexrow="0" isguessbutton="true"></colorbutton-stub>
    <colorbutton-stub color="lightgray" index="1" indexrow="0" isguessbutton="true"></colorbutton-stub>
    <colorbutton-stub color="lightgray" index="2" indexrow="0" isguessbutton="true"></colorbutton-stub>
    <colorbutton-stub color="lightgray" index="3" indexrow="0" isguessbutton="true"></colorbutton-stub>
  </div>
  <pegslist-stub numslots="4" indexrow="0"></pegslist-stub>
</div>
`;
